mlir/lib/Dialect/Rock/utility/loweringUtils.cpp

```diff
@@ -125,3 +128,225 @@ Type mlir::rock::vectorTypeOrSelf(Type elementType, int64_t len) {
     return elementType;
   return VectorType::get({len}, elementType);
 }
+
+static SmallVector<unsigned> getOutDimIndices(int64_t outDimStartIdx,
+                                              int64_t newDimSize) {
+  SmallVector<unsigned> outDims;
+  for (int i = 0; i < newDimSize; i++) {
+    outDims.push_back(outDimStartIdx + i);
+  }
+  return outDims;
+}
+
+static std::tuple<SmallVector<StringRef>, SmallVector<int64_t>>
+getCombinedOrder(const LowerDimPartInfo &lhs, const LowerDimPartInfo &rhs) {
```

Comment:
I'm very suspicious of the fact that you need this.

Reply:
The fact this was needed to factor out create similiar views for 1) loaded tile from global and 2) reshuffled tile from global where the difference is only how iter maps to indices in regs.